### PR TITLE
Adjust katello-nightly-release to use ci.centos.org when testing inst…

### PR DESCRIPTION
**Pull request by zjhuntin**
…allation tests

### CODE REVIEW

**Comment by zjhuntin** [comment]
Requires: https://github.com/theforeman/foreman-infra/pull/698

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katelloRelease.groovy [hunk 1]
```diff
@@ -60,33 +60,12 @@ pipeline {
...
-                    sh "scp -F ssh_config centos7-katello-bats-ci:/root/foreman-debug.tar.xz debug/ || true"
-                    sh "scp -F ssh_config centos7-katello-bats-ci:/var/log/foreman-installer/katello.log debug/ || true"
-
-                    sh "vagrant destroy centos7-katello-bats-ci"
-
-                    archive "debug/*"
-                    script { step([$class: "TapPublisher", testResults: "debug/*.tap"]) }
-                    deleteDir()
+                withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'jenkins-centos', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME']]) {
+                    runPlaybook('ci/centos.org/ansible/run_jenkins_job.yml', 'localhost', "[jenkins_job_name=foreman-nightly-test, jenkins_username=${env.USERNAME}, jenkins_password=${env.PASSWORD}]"
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katelloRelease.groovy]: The name here is `foreman-nightly-test` but the pipeline is for katello. That doesn't sound right.

**Comment by zjhuntin** [comment]
- Do we archive the debug logs somewhere?

Yes, they will be archived on the centos job this will kick off. It will archive a sosreport, foreman debug, and bats tests results. I was unsure how to link the centos job with this outside of looking in the console for it, but we probably can have some way to reach it directly from a job.

- Do we publish the test results somewhere?

The tests will be freely available to see on the centos jenkins job that this will start, so that anyone can see it. The job being this one: https://github.com/theforeman/foreman-infra/pull/687, once it has been published.

- How would I replicate this to debug the issue? (either as packager or developer)

We are not changing the pipeline that is being run, more trying to outsource it to hardware that will be able to run it without any memory issues. A user can see from the connected job what it is being run. I admit that it abstracted, but they would be able to follow the instructions from the centos job.

**Comment by zjhuntin** [comment]
I agree that it is difficult to see when a job fails and then following a workflow to replicate it. I think that perhaps we should take a different strategy with this then. Instead of trying to have them read the job from any Jenkins to try and follow the workflow, which is by necessity going to do some things you will not have to do locally, to instead write out how these can be replicated locally. This could be through solved through a couple of ways off the top of my head.

- Write out how to replicate these important jobs locally using forklift. We could host these on theforeman.org and then link each job to their page.

- Write docker compose files to run these locally, the biggest advantage here being they just have to get docker working, and not have have libvirt, vagrant etc to run it.

**Comment by ehelms** [comment]
For me, this is the same as previous from a developer perspective except what has changed is instead of `vagrant up centos7-katello-bats-ci` it is now 'ansible-playbook pipelines/pipeline_katello_nightly.yml`. That aspect we can communicate more clearly through Forklift docs and potentially infra docs.

**Comment by ekohl** [comment]
@zjhuntin a documented way of debugging these issues on theforeman.org sounds like great idea.

**Comment by ehelms** [comment]
I thnk thats a good idea, but not a blocker given this does not change the status quo with respect to debugging.

**Pull request merged**